### PR TITLE
remove local provisioner daemonset, no longer used

### PR DESCRIPTION
**Pull request by dobsonj**
cleanup work following up on https://github.com/openshift/local-storage-operator/pull/257#discussion_r670922916
/assign @sp98 
/assign @rohantmp 
/assign @gnufied

### CODE REVIEW

File: common/names.go [hunk 1]
```diff
@@ -10,7 +10,6 @@ import (
 
 const (
 	defaultDiskMakerImageVersion = "quay.io/openshift/origin-local-storage-diskmaker"
-	defaultProvisionImage        = "quay.io/openshift/origin-local-storage-static-provisioner"
```
Comment (sp98) [inline common/names.go:13]: We also have references to this image in some other places as well.  Like  `config/manager/manager.yaml` , `config/manifest/4.9/image-references` and in the csv files.

**Review by sp98 — CHANGES_REQUESTED** [review]

File: common/names.go [hunk 2]
```diff
@@ -56,14 +54,6 @@ const (
 	DiscoveryMetricsServingCert = "discovery-metric-serving-cert"
 )
 
-// GetLocalProvisionerImage return the image to be used for provisioner daemonset
-func GetLocalProvisionerImage() string {
-	if provisionerImageFromEnv := os.Getenv(ProvisionerImageEnv); provisionerImageFromEnv != "" {
```
Comment (sp98) [inline common/names.go:61]: `ProvisionerImageEnv` reference should be removed in the `names.go` file.

**Review by sp98 — CHANGES_REQUESTED** [review]

File: common/names.go (hunk 1 above)
Comment (dobsonj) [reply common/names.go:13]: Thank @sp98, updated.

File: common/names.go (hunk 2 above)
Comment (dobsonj) [reply common/names.go:61]: Done

**Review by sp98 — APPROVED** [review]

**Pull request merged**